client/dashboard/sites/site/environment-switcher.tsx

```diff
@@ -0,0 +1,167 @@
+import { useQuery, useMutation } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	Button,
+	Dropdown,
+	MenuGroup,
+	MenuItem,
+	Spinner,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { sprintf, __ } from '@wordpress/i18n';
+import { Icon, chevronDownSmall, plus } from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+import { siteByIdQuery } from '../../app/queries/site';
+import { stagingSiteCreateMutation } from '../../app/queries/site-staging-sites';
+import { production, staging } from '../../components/icons';
+import RouterLinkMenuItem from '../../components/router-link-menu-item';
+import { canManageSite, canCreateStagingSite } from '../features';
+import type { Site } from '../../data/types';
+
+type EnvironmentType = 'production' | 'staging';
+
+const Environment = ( { env }: { env: EnvironmentType } ) => {
+	if ( env === 'staging' ) {
+		return (
+			<HStack justify="flex-start" style={ { width: 'auto', flexShrink: 0 } }>
+				<Icon icon={ staging } />
+				<span>{ __( 'Staging' ) }</span>
+			</HStack>
+		);
+	}
+
+	return (
+		<HStack justify="flex-start" style={ { width: 'auto', flexShrink: 0 } }>
+			<Icon icon={ production } />
+			<span>{ __( 'Production' ) }</span>
+		</HStack>
+	);
+};
+
+const CurrentEnvironment = ( { site }: { site: Site } ) => {
+	if ( site.is_wpcom_staging_site ) {
+		return <Environment env="staging" />;
+	}
+
+	return <Environment env="production" />;
+};
+
+const EnvironmentSwitcherDropdown = ( {
```

Comment:
Nit: is it possible to pre-select the current environment when opening the dropdown? So the below image should preselect the Staging one.

<img width="267" height="179" alt="Image" src="https://github.com/user-attachments/assets/79a012a6-2b80-4903-8bcb-7560e0927c08" />

Reply:
The regular behaviour is to have a dot to the left of the label to indicate the selected item. We can’t do that here due to the environment icon though.

<img width="343" height="53" alt="Screenshot 2025-07-14 at 09 45 22" src="https://github.com/user-attachments/assets/342e277f-0249-4465-87c9-4ef7d9f36db4" />

It’s generally recommended to give focus to the first focusable element in a menu to maintain accessibility. I don’t think we should change that behaviour as it could confuse some users that rely on assistive technologies.

Reply:
TIL. Okay! 👍